src/provision.js

```diff
@@ -296,6 +296,21 @@ class Provisioner {
     // any particular worker type
     forSpawning = shuffle.knuthShuffle(forSpawning);
 
+    // Let's find the unique worker types in the list of requests to make
+    let toSpawnWorkerTypes = _.intersection(forSpawning.map(x => x.workerType.workerType));
+
+    let disabled = [];
+    for (let toTest of workerTypes.filter(x => toSpawnWorkerTypes.indexOf(x) !== -1)) {
+      // Consider iterating over a list where we filter out those worker types
+      // which don't have outstanding requests
+      let canLaunch = await this.awsManager.workerTypeCanLaunch(toTest);
+      if (!canLaunch) {
+        disabled.push(toTest.workerType);
+      }
```

Comment:
Getting rid of `disabled` would certainly help... 

I wish I saw a more elegant way to do this... but it seems okay to me.